### PR TITLE
feat(forms): Collapse previous array items on add

### PR DESCRIPTION
**Pull request by talend-glorieux**
**What is the problem this PR is trying to solve?**
Adding elements to an array forces the user to scroll to fill the newly added item.

**What is the chosen solution to this problem?**
Collapse all previous items when adding a new one.

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

### CODE REVIEW

File: packages/forms/src/fields/ArrayField.js [hunk 1]
```diff
@@ -199,6 +198,11 @@ class ArrayField extends Component {
 		if (isFixedItems(schema) && allowAdditionalItems(schema)) {
 			itemSchema = schema.additionalItems;
 		}
+		formData.map((item) => {
+			const i = item || {};
+			i.isClosed = true;
+			return i;
+		});
```
Comment (jmfrancois) [inline packages/forms/src/fields/ArrayField.js]: please use forEach instead of map

File: packages/forms/src/fields/ArrayField.js [hunk 2]
```diff
@@ -199,6 +198,11 @@ class ArrayField extends Component {
 		if (isFixedItems(schema) && allowAdditionalItems(schema)) {
 			itemSchema = schema.additionalItems;
 		}
+		formData.map((item) => {
+			const i = item || {};
+			i.isClosed = true;
```
Comment (jmfrancois) [inline packages/forms/src/fields/ArrayField.js]: create a const just to assign a value if exists ?
i && i.isClosed = true 
_.set(i, 'isClosed', true)

File: packages/forms/src/templates/ArrayFieldTemplate.js [hunk 3]
```diff
@@ -57,23 +57,24 @@ if (process.env.NODE_ENV !== 'production') {
...
 					name="btn-new-element"
 					disabled={items.length >= maxItems}
 					onClick={onAddClick}
 				>
 					{`NEW ${props.type}`}
 				</button>}
+			{items &&
+				items.map(element =>
+					<FieldTemplate element={element} cantDelete={items.length <= minItems} />
+				)}
```
Comment (jmfrancois) [inline packages/forms/src/templates/ArrayFieldTemplate.js]: map need key

File: packages/forms/src/templates/ArrayFieldTemplate.scss [hunk 4]
```diff
@@ -1,6 +1,11 @@
 $tf-filterwidget-shadow-color: $shadow !default;
 
 .ArrayFieldTemplate {
+
```
Comment (jmfrancois) [inline packages/forms/src/templates/ArrayFieldTemplate.scss]: usless return, please remove

File: packages/forms/src/templates/ArrayFieldTemplate.js (hunk 3 above)
Comment (talend-glorieux) [reply packages/forms/src/templates/ArrayFieldTemplate.js]: eslint doesn't require it.

Comment (jmfrancois) [reply packages/forms/src/templates/ArrayFieldTemplate.js]: but you know we should do it, may be a false positive.

**Pull request merged**